plugins/module_utils/network/iosxr/rm_templates/bgp_templates.py

```diff
@@ -0,0 +1,2813 @@
+    if conf:
+        command = "maximum-prefix"
+        if "max_limit" in conf:
+            command += " " + str(conf["max_limit"])
+        if "threshold_value" in conf:
+            command += " " + str(conf["threshold_value"])
+        if "restart" in conf:
+            command += " restart " + str(conf["restart"])
+        elif "warning_only" in conf:
+            command += " warning-only"
+        elif "discard_extra_paths" in conf:
+            command += " discard-extra-paths"
+
+    return command
+
+
+def _tmpl_soft_reconfiguration(config_data):
+    conf = config_data.get("soft_reconfiguration", {})
+    if conf:
+        command = "soft-reconfiguration "
+        if "inbound" in conf:
+            command += "inbound"
+            if "set" in conf["inbound"]:
+                pass
+            elif "always" in conf["inbound"]:
+                command += " always"
+            if "inheritance_disable" in conf["inbound"]:
+                command += " inheritance-disable"
+
+    return command
+
+
+def _tmpl_remove_private_AS(config_data):
+    conf = config_data.get("remove_private_AS", {})
+    if conf:
+        command = "remove-private-AS"
+        if "inbound" in conf:
+            command += " inbound"
+        if "entire_aspath" in conf:
+            command += " entire-aspath"
+        if "inheritance_disable" in conf:
+            command += " inheritance-disable"
+    return command
+
+
+def _templ_local_as(config_data):
+    conf = config_data.get("local_as", {})
+    if conf.get("value"):
+        command = "local-as " + str(conf.get("value", {}))
+    if "no_prepend" in conf:
+        command = "local-as"
+        if "replace_as" in conf.get("no_prepend", {}):
+            if "dual_as" in conf.get("no_prepend", {}).get("replace_as", {}):
+                command += " no-prepend replace-as dual-as"
+            elif "set" in conf.get("no_prepend", {}).get("replace_as", {}):
+                command += " no-prepend replace-as"
+        elif "set" in conf.get("no_prepend", {}):
+            command += " no-prepend"
+    return command
+
+
+class Bgp_templatesTemplate(NetworkTemplate):
+    def __init__(self, lines=None, module=None):
+        super(Bgp_templatesTemplate, self).__init__(
+            lines=lines,
+            tmplt=self,
+            module=module,
+        )
+
+    # fmt: off
+    PARSERS = [
+        {
+            "name": "router",
+            "getval": re.compile(
+                r"""
+                ^router\s
+                bgp
+                \s(?P<as_num>\S+)
+                $""",
+                re.VERBOSE,
+            ),
+            "setval": "router bgp {{ as_number }}",
+            "compval": "as_number",
+            "result": {"as_number": "{{ as_num }}"},
+            "shared": True,
+        },
+        {
+            "name": "address_family",
+            "getval": re.compile(
+                r"""
+                \s+(?P<nbr_address>neighbor-group\s\S+)
```

Comment:
```suggestion
                \s+neighbor-group\s(?P<nbr_address>\S+)
```
The split in every regex won't be needed!

Reply:
ack